emoji_fzf/emoji_fzf.py

```diff
@@ -28,13 +29,34 @@ def cli():
     default=False,
     show_default=True,
 )
-def preview(prepend_emoji=False):
+@click.option(
+    "-c",
+    "--custom-aliases",
+    "custom_aliases_file",
+    help="Path to custom alias JSON file",
+    type=click.File("r"),
+    default=None,
+)
+def preview(prepend_emoji=False, custom_aliases_file=None):
     """Return an fzf-friendly search list for emoji"""
+    custom_aliases = {}
+    if custom_aliases_file:
+        try:
+            custom_aliases = {
+                x.get("emoji"): x.get("aliases") for x in json.load(custom_aliases_file)
```

Comment:
Instead maybe update the `EMOJIS` dictionary with the new aliases here?
Doing it in the loop below adds about 10ms overhead when there's no `custom_aliases_file` specified.

master:
```
❯ hyperfine -w3 -r10 'emoji-fzf preview > /dev/null'
Benchmark #1: emoji-fzf preview > /dev/null
  Time (mean ± σ):      87.8 ms ±   3.1 ms    [User: 82.1 ms, System: 5.4 ms]
  Range (min … max):    84.7 ms …  95.3 ms    10 runs
```

This patch:
```
❯ hyperfine -w3 -r10 'emoji-fzf preview > /dev/null'
Benchmark #1: emoji-fzf preview > /dev/null
  Time (mean ± σ):      96.1 ms ±   8.8 ms    [User: 85.9 ms, System: 9.8 ms]
  Range (min … max):    88.2 ms … 117.4 ms    10 runs
```

Reply:
To really improve performance I think the best way forward is to update the `EMOJIS` dict structure to use the actual emojis as keys. That'll make lookups way faster than what can be done now. BTW is there a particular reason why the `aliases` key holds a `set` rather than a list? Performance?

Structure proposal:

```python
EMOJIS = {'😀': {'name': 'grinning_face', 'aliases': ['happy', ':D', 'joy']} ... }
```

Reply:
Hm. The customization JSON schema could also be simplified to something like that:

```json
[
  "👍": ["cust-alias1", "cust-alias2"],
  "🌟": ["cust-alias3"]
]
```

It's a bit more straightforward and easier to write.

Or we could do the user-unfriendly thing and have it follow the exact same schema as the `EMOJIS` dict - which would make merging the two dicts trivial (and faster?). I personally prefer the former, simpler approach.

Reply:
Former looks good to me, I agree it's nicer :D

Reply:
> To really improve performance I think the best way forward is to update the `EMOJIS` dict structure to use the actual emojis as keys. That'll make lookups way faster than what can be done now. BTW is there a particular reason why the `aliases` key holds a ~`set`~ `tuple` rather than a list? Performance?
> 
> Structure proposal:
> 
> ```python
> EMOJIS = {'😀': {'name': 'grinning_face', 'aliases': ['happy', ':D', 'joy']} ... }
> ```

The `EMOJIS` dict is keyed by canonical name for the `get()` function to grab it. BTW, feel free to just modify `EMOJIS` in place instead of attaching it to a value on `ctx`, then you don't have to pass it through to the sub commands. Should make the patch smaller.

Reply:
> Former looks good to me, I agree it's nicer :D

Implemented in 89f21dc

Reply:
> BTW, feel free to just modify `EMOJIS` in place instead of attaching it to a value on `ctx`, then you don't have to pass it through to the sub commands. Should make the patch smaller.

Implemented in 637a906